vtex-sites/starter.store
# chore: allows daily dependabot version updates for @faststore/* deps

### The build (and CICD tasks) depends on https://github.com/vtex-sites/starter.store/pull/12

## What's the purpose of this pull request?

This PR is the first step of the FastStore Evergreen Dependabot.
<img width="713" alt="Screen Shot 2022-12-20 at 18 49 16" src="https://user-images.githubusercontent.com/11325562/208773774-f0132a80-f2f8-4735-b9aa-e1e29a9e2999.png">


This PR aims to enable daily [dependabot version updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates) for the `@faststore/*` dependencies.

We can deal with dependabot at first, but after we can try other multi-platform tools for automated dependency updates, like [renovate](https://docs.renovatebot.com/#supported-platforms).

## How does it work?

Based on the Github Docs previously mentioned. We can enable the version updates by creating the `dependabot.yml` file in the `.github` folder with [custom configs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file).

GitHub also allows us to enable `dependabot` version updates directly from the settings, but this is enable once we upload the `dependabot.yml` file, as mentioned previously.
<img width="1444" alt="Screen Shot 2022-12-13 at 20 50 59" src="https://user-images.githubusercontent.com/11325562/207470408-7c13e0dd-db32-4dc5-8b08-31cf235385de.png">

## How to test it?

For testing purposes, We created a new lib called [@test-fs/test-gh-dependabot-updates](http://www.npmjs.com/package/@test-fs/test-gh-dependabot-updates), and used it as a dependency in the [vtexgame1.store](https://github.com/vtex-sites/vtexgame1.store/blob/main/.github/dependabot.yml) repo. 
After that, we add a similar `dependabot` config file there and observed that `dependabot` opens the PRs as expected.
PRs opened by dependabot:
https://github.com/vtex-sites/vtexgame1.store/pull/16
https://github.com/vtex-sites/vtexgame1.store/pull/15
https://github.com/vtex-sites/vtexgame1.store/pull/14
https://github.com/vtex-sites/vtexgame1.store/pull/13

Why do we use other repos and libs?
<strike>- Webops doesn't support builds from the Evergreen approach using `.faststore` temporary folder.</strike> now it works after https://github.com/vtex-sites/starter.store/pull/5

## References
https://github.blog/changelog/2022-06-02-enable-dependabot-version-updates-from-the-repository-settings-page/
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
https://www.npmjs.com/package/@faststore/core
https://docs.renovatebot.com/#supported-platforms
